frontend/src/components/Sidebar/Sidebar.tsx

```diff
@@ -62,9 +70,22 @@ const SidebarComponent: React.FC<Props> = ({ schedule, major }) => {
   );
 };
 
+const SidebarComponent: React.FC<SidebarProps> = ({ schedule, major }) => {
+  const majorObj: Major | undefined = useSelector<AppState, Major | undefined>(
+    state =>
+      state.majorState.majors.find((majorObj: Major) => majorObj.name === major)
+  );
+
+  return majorObj ? (
+    <MajorSidebarComponent schedule={schedule} major={majorObj} />
+  ) : (
+    <NoMajorSidebarComponent />
+  );
+};
+
 const mapStateToProps = (state: AppState) => ({
   schedule: getScheduleFromState(state),
-  major: getDeclaredMajorFromState(state),
+  major: getScheduleDataFromState(state).major,
```

Comment:
Fetch major string from schedule state instead of user.

Reply:
Could we make a `getMajorFromState` function for this? Seems like we would use it a lot down the road.

Reply:
That would make sense, I'll add that for major and coop cycle.